index.js

```diff
@@ -106,43 +113,53 @@ var S3Sync = klass(function (config, options) {
 
       debug('putting original file %s at destination %s', file, s3FileNameWithPrefix)
 
-      this.client.putFile(
-        file,
-        s3FileNameWithPrefix,
-        _.extend({}, this.getSettings(), this._mergeHeaders(file)),
-        this.md5PreCheck.bind(this, file, md5File, done)
-      )
-      .on('error', function (e) {
-        console.error('error in putting original file', e)
+      var body =  fs.createReadStream(file)
+      var headers = _.extend({}, this.getSettings(), this._mergeHeaders(file))
+
+      this.client.upload({
+        Body: body,
+        Bucket: this.bucket,
+        Key: s3FileNameWithPrefix,
+        headers
       })
+      .send(function(err, data) {
+        if (err) {
+          console.error('error in putting original file', e)
+        } else {
+          this.md5PreCheck(file, md5File, done)
+        }
+      }.bind(this))
     }
   , md5PreCheck: function (file, md5File, done) {
       this.s3FilePreCheck(md5File)
         .then(function () {
           debug('putting new file', md5File)
-          this.client.putFile(
-            file,
-            md5File,
-            _.extend({}, this.getSettings(), this._mergeHeaders(file)),
-            done
-          )
-          .on('error', function (e) {
-            console.error('error in putting new file', e)
+          var md5body =  fs.createReadStream(file)
+          var md5headers = _.extend({}, this.getSettings(), this._mergeHeaders(md5File))
+          this.client.upload({
+            Body: md5body,
+            Bucket: this.bucket,
+            Key: md5File,
+            md5headers
```

Comment:
Hm, yes md5headers would be stored as an object. crap. I keep forgetting about lodash. Really need to keep that in mind whenever dealing with objects. _.merge is so handy.